src/App.jsx

```diff
@@ -14,33 +17,98 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+export const App = () => {
+  const [visibleGoods, setVisibleGoods] = useState(goodsFromServer);
+  const [showReversed, setShowReversed] = useState(false);
+  const [sortedBy, setSortedBy] = useState('');
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+  const applyReverse = data => (showReversed ? data.reverse() : data);
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+  const processData = (data, sort) => {
+    setVisibleGoods(applyReverse(data));
+    setSortedBy(sort);
+  };
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+  const sortByAlphabet = () => {
+    processData(
+      [...goodsFromServer].sort((a, b) => a.localeCompare(b)),
+      'alphabet',
+    );
+  };
+
+  const sortByLength = () => {
+    processData(
+      [...goodsFromServer].sort(
+        (a, b) => a.length - b.length || a.localeCompare(b),
+      ),
+      'length',
+    );
+  };
+
+  const toggleReverse = () => {
+    setShowReversed(!showReversed);
+    setVisibleGoods([...visibleGoods].reverse());
+  };
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={classNames('button', 'is-info', {
+            'is-light': sortedBy !== 'alphabet',
+          })}
+          onClick={() => {
+            sortByAlphabet();
+          }}
+        >
+          Sort alphabetically
+        </button>
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+        <button
+          type="button"
+          className={classNames('button', 'is-success', {
+            'is-light': sortedBy !== 'length',
+          })}
+          onClick={() => {
+            sortByLength();
+          }}
+        >
+          Sort by length
+        </button>
+
+        <button
+          type="button"
+          className={classNames('button', 'is-warning', {
+            'is-light': !showReversed,
+          })}
+          onClick={toggleReverse}
+        >
+          Reverse
+        </button>
+
+        {sortedBy === '' && showReversed === false ? null : (
```

Comment:
`const isResetButtonVisible = sortedBy || isReversed;`
```suggestion
        {isResetButtonVisible && (// your button)}
```
- You don't need to compare with boolean values or with an empty string https://developer.mozilla.org/en-US/docs/Glossary/Falsy
- And you don't need to use a ternary operator in this case, as you don't render any other UI in this case